syntax = "proto3";

package aom.iot.device;

import "action.proto";

enum IO {
    INPUT = 0;
    OUTPUT = 1;
}

enum Signal {
    DIGITAL = 0;
    ANALOG = 1;
}

message Device {
    string name = 1;
    string uuid = 2;
    string token = 3;
    string user_uuid = 4;
    string client_uuid = 5;
    string client_name = 6;
    uint32 number = 7;
    IO io = 8;
    Signal signal = 9;
    repeated aom.iot.action.Action actions = 10;
}
